src/main/scala/PropertyChecker.scala

```diff
@@ -0,0 +1,483 @@
+package CSVValidation
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.fasterxml.jackson.databind.node._
+
+import scala.collection.mutable.HashMap
+import scala.jdk.CollectionConverters._
+import scala.util.matching.Regex
+
+
+object PropertyChecker {
+  val startsWithUnderscore = "^_:.*".r
+  val containsColon = ".*:.*".r
+  val mapper = new ObjectMapper
+  val invalidValueWarning = "invalid_value"
+  val Bcp47Regular = "(art-lojban|cel-gaulish|no-bok|no-nyn|zh-guoyu|zh-hakka|zh-min|zh-min-nan|zh-xiang)"
+  val Bcp47Irregular = "(en-GB-oed|i-ami|i-bnn|i-default|i-enochian|i-hak|i-klingon|i-lux|i-mingo|i-navajo|i-pwn|i-tao|i-tay|i-tsu|sgn-BE-FR|sgn-BE-NL|sgn-CH-DE)"
+  val Bcp47Grandfathered = "(?<grandfathered>" + Bcp47Irregular + "|" + Bcp47Regular + ")"
+  val Bcp47PrivateUse = "(x(-[A-Za-z0-9]{1,8})+)"
+  val Bcp47Singleton = "[0-9A-WY-Za-wy-z]"
+  val Bcp47Extension = "(?<extension>" + Bcp47Singleton + "(-[A-Za-z0-9]{2,8})+)"
+  val Bcp47Variant = "(?<variant>[A-Za-z0-9]{5,8}|[0-9][A-Za-z0-9]{3})"
+  val Bcp47Region = "(?<region>[A-Za-z]{2}|[0-9]{3})"
+  val Bcp47Script = "(?<script>[A-Za-z]{4})"
+  val Bcp47Extlang = "(?<extlang>[A-Za-z]{3}(-[A-Za-z]{3}){0,2})"
+  val Bcp47Language = "(?<language>([A-Za-z]{2,3}(-" + Bcp47Extlang + ")?)|[A-Za-z]{4}|[A-Za-z]{5,8})"
+  val Bcp47Langtag = "(" + Bcp47Language + "(-" + Bcp47Script + ")?" + "(-" + Bcp47Region + ")?" + "(-" + Bcp47Variant + ")*" + "(-" + Bcp47Extension + ")*" + "(-" + Bcp47PrivateUse + ")?" + ")"
+  val Bcp47LanguagetagRegExp: Regex = ("^(" + Bcp47Grandfathered + "|" + Bcp47Langtag + "|" + Bcp47PrivateUse + ")").r
+
+
+  val StringDataTypes = Array[String](
```

Comment:
i was more thinking you could have one `Constants` object containing all of the constancts (including the ones below).